internal/pkg/aws/sessions/sessions.go

```diff
@@ -104,6 +112,12 @@ func (p *Provider) FromProfile(name string) (*session.Session, error) {
 	if aws.StringValue(sess.Config.Region) == "" {
 		return nil, &errMissingRegion{}
 	}
+	if _, err := p.sessionValidator.ValidateCredentials(sess); err != nil {
+		if strings.Contains(err.Error(), "context deadline exceeded") || strings.Contains(err.Error(), "NoCredentialProviders") {
+			return nil, &errCredProviderTimeout{name}
```

Comment:
If we return `errCredProviderTimeout` when the real error is "NoCredentialProviders", we lose some troubleshooting information. For example, we will print the error `"retrieving credentials timeout"` when in reality the creds didn't timeout, they just quickly failed to fetch.

Can we differentiate between the two?

```go
   if isCredRetrievalErr(err):
      return nil, &errCredRetrieval{profile: name, parentErr: err}
```

and we have the following instead of `errCredProviderTimeout`:
```go
type errCredRetrieval struct {
	profile string
	parentErr error
}
func (e *errCredRetrieval) Error() string {
	return e.parentErr
}
func (e *errCredRetrieval) RecommendActions() string {
        // same as before
}
```
```go
func isCredRetrievalErr(err error) bool {
    return strings.Contains(err.Error(), "context deadline exceeded") || strings.Contains(err.Error(), "NoCredentialProviders")
}
```

Reply:
This makes sense to me.